typedmodels/models.py

```diff
@@ -81,7 +81,11 @@ class Meta:
             declared_fields = dict((name, element) for name, element in classdict.items() if isinstance(element, Field))
 
             for field_name, field in declared_fields.items():
-                field.null = True
+                # In Django 1.8+, warnings will be triggered by the system
+                # check for M2M fields setting if we set null to True. Prevent
+                # those warnings by setting null only for non-M2M fields.
+                if django.VERSION < (1, 8) or not field.many_to_many:
+                    field.null = True
```

Comment:
> Setting many to many fields null had no effect before Django 1.8, correct?

That is correct and this PR leaves the Django < 1.8 behavior the same. The reason for the Django version test is that `many_to_many` is an attribute that was added to `django.db.models.fields.Field` in Django 1.8. I'm hoping Django 1.7 is not long for this world so hopefully this can merge as is and then the Django version conditional can be removed when 1.7 support is dropped.